### PR TITLE
Updated code for UAT TEST PLAN

### PR DESCRIPTION
**Pull request by bgfoshay**
## What’s changing?

_For External API changes, add the changelog tag to the PR and some CHANGELOG text here_
added new test to the API automation Framework

## Why?

_Link to specific issue (ex: Closes NAC-123)_

## This PR has:

- [ ] **JIRA ticket(s)** (_Must be listed in PR request and branch name should follow convention NAC-123-description_)<description>

- [ ] **Security Related Changes** (_Security Tab questions have been answered on JIRA ticket(s)_)

- [ ] **Commented source code** (_required on public classes and methods, and elsewhere as appropriate_)

- [ ] **Documented API changes** (_required on public classes and methods, and elsewhere as appropriate_)
    - [ ] External API documentation - OpenApi docs and Generated Docs (e.g. https://github.com/18F/piipan/tree/dev/docs/openapi) 
    - [ ] Internal API documentation (e.g. https://github.com/18F/piipan/tree/dev/metrics/docs/openapi)   

- [ ] **Developer documentation** (_for new build/test/API changes or complex portions of the system_)

- [x] **Automated unit tests** (_to maintain or increase level of code coverage_)

- [ ] **Changes to IAC** (_add any deployment steps that require manual intervention to the draft release notes_)
    - [ ] Have IaC scripts run successfully against an existing deployed environment?
    - [ ] Have IaC scripts run successfully against a newly deployed environment?
    - [ ] Have any new bash scripts been given the appropriate permissions? (e.g. git update-index --chmod=+x $script)
    - [ ] Do the IaC changes require deleting existing resources?

## Anything else?

_Add other details that are helpful for review here_

## Code Review Checklist:

- [ ] Automated checks all pass? (code builds successfully, tests pass, test coverage meets/exceeds threshold, etc.) 
- [ ] Have all requirements and Acceptance Criteria been met?
- **If there are code changes**
    - [ ] Is there no unnecessary complexity? Code is readable, maintainable, etc.
    - [ ] Do the changes follow SOLID principles?
    - [ ] Are the changes robust? Concurrency considered? Performance? Security (e.g. no sql injections)?
    - [ ] Are newly added dependencies needed and appropriate? Are dependency files (e.g. packages.json & packages.lock.json) updated?
    - [ ] For any new projects/solutions, have they been added to code coverage files?
    - [ ] Public classes & methods are documented    
- **If there are API changes**
    - [ ] Does API make sense? Useful and concise?
    - [ ] Are existing API interfaces/contracts intact? i.e. no breaking changes
    - [ ] Are API tests included?
    - [ ] Are API responses appropriate and documented? (200s, 201, 404s, etc.)
    - [ ] Are internals kept private and not leaked into the API?
    - [ ] External API documentation - OpenApi docs and Generated Docs 
    - [ ] Internal API documentation    
- **If there are IAC changes**
    - [ ] IAC Scripts are readable & maintainable
    - [ ] IAC Scripts are documented
    - [ ] IAC scripts pass required checks
    - [ ] IAC scripts were run successfully by author
- **General Logging Questions**
    - [ ] Do new capabilities (or changed capabilities) include logging?
    - [ ] Is PII not logged?
- **General Error Handling Questions**
    - [ ] Has exception handling been added for new capabilities (or changed capabilities)?
    - [ ] Are errors/exceptions properly communicated to end-users? Were there requirements for user facing error messages?
    - [ ] Are errors/exceptions logged?
    - [ ] Is PII kept out of error/exception logging?
- **If there are Developer Documentation Changes**
    - [ ] General Project documentation (under piipan\docs. e.g. iac.md, architecture.md, etc.)
    - [ ] Github documentation additions/changes (Readmes)
- **General automated test questions**
    - [ ] End-to-End Tests provided (if code implements new user story)
    - [ ] Are tests sufficient?
    - [ ] Corner cases considered?
    - [ ] Are performance tests needed?

### CODE REVIEW

File: api-tests/functional-testing/fns_nac_automation-qa_env.json [hunk 1]
```diff
@@ -1,10 +1,10 @@
 {
-	"id": "5002971a-9989-47d5-af9e-04007975c110",
-	"name": "FNS_DEV_TEST_ENV",
+	"id": "07dcc5b2-9b3d-4b9d-a6d4-9ef459e47b92",
+	"name": "FNS_QA_Env",
 	"values": [
 		{
 			"key": "base_url",
-			"value": "https:///fns-apim-duppartapi-dev.azure-api.us",
+			"value": "https:///fns-apim-duppartapi-test.azure-api.us ",
```
Comment (ccaprio97) [inline api-tests/functional-testing/fns_nac_automation-qa_env.json:7]: nickpick non-blocking: should the space be here or does it not matter. Also should the url have 3 / after https also not sure if it matters

File: api-tests/functional-testing/fns_nac_automation-qa_env.json [hunk 2]
```diff
@@ -15,69 +15,63 @@
...
+			"value": "eb",
 			"type": "default",
 			"enabled": true
 		},
 		{
-			"key": "stateEC",
-			"value": "ec",
-			"type": "default",
+			"key": "eb_uploadKey",
+			"value": "",
```
Comment (ccaprio97) [inline api-tests/functional-testing/fns_nac_automation-qa_env.json:43]: question/comment: not 100% how this is working. also curious why some of these have values and others dont but could just be knowledge gap

Comment (ccaprio97) [reply api-tests/functional-testing/fns_nac_automation-qa_env.json:43]: or just a null test?

File: api-tests/functional-testing/fns_nac_automation-qa_env.json [hunk 3]
```diff
@@ -15,69 +15,63 @@
...
-			"enabled": false
+			"type": "secret",
+			"enabled": true
 		},
 		{
-			"key": "statusCode",
-			"value": "",
-			"type": "any",
+			"key": "search_reason",
+			"value": "RecerTificatio",
```
Comment (ccaprio97) [inline api-tests/functional-testing/fns_nac_automation-qa_env.json:73]: nitpick: should recertification have the "n" at the end?

File: api-tests/functional-testing/fns_nac_automation-qa_env.json [hunk 4]
```diff
@@ -87,19 +81,19 @@
...
 			"type": "any",
 			"enabled": true
 		}
 	],
 	"_postman_variable_scope": "environment",
-	"_postman_exported_at": "2022-11-11T19:19:48.754Z",
-	"_postman_exported_using": "Postman/10.0.42"
+	"_postman_exported_at": "2023-01-25T17:44:34.186Z",
+	"_postman_exported_using": "Postman/10.8.6"
 }
```
Comment (ccaprio97) [inline api-tests/functional-testing/fns_nac_automation-qa_env.json:99]: nitpick non-blocking: I believe git files should be pushed up with a new line at the end of the files typically

File: api-tests/functional-testing/generateRandomDataFile.js [hunk 5]
```diff
@@ -7,4 +8,6 @@ const parsed = parse(file, {
   skip_empty_lines: true
 })
 const chosenRecord = parsed[Math.floor(Math.random() * parsed.length)]
+const rowLength = parsed.length;
+chosenRecord["Total_Row_Count"]=rowLength;
 fs.writeFileSync(outputFilename, JSON.stringify([chosenRecord]))
```
Comment (ccaprio97) [inline api-tests/functional-testing/generateRandomDataFile.js:13]: nitpick non blocking: again I think git files are supposed to be pushed up with a new line at the end

File: api-tests/functional-testing/fns_nac_automation-postman_collection.json [hunk 6]
```diff
@@ -79,58 +82,277 @@
...
+						"type": "text/javascript"
+					}
+				},
+				{
+					"listen": "prerequest",
+					"script": {
+						"exec": [
+							"pm.request.removeHeader(\"Ocp-Apim-Subscription-Key\");\r",
+							"pm.request.removeHeader(\"X-Initiating-State\");\r",
+							"// pm.request.headers.add({key: \"Ocp-Apim-Subscription-Key\", value: pm.environment.get(\"findMatch_SubKey\")});\r",
```
Comment (dborgen2) [inline api-tests/functional-testing/fns_nac_automation-postman_collection.json:165]: Question: Is this commented line needed? I noticed this was also in some other existing Pre-request scripts. If it's needed for something, can we include a comment about how to use it?

File: api-tests/functional-testing/fns_nac_automation-qa_env.json (hunk 1 above)
Comment (mhaque-ventera) [reply api-tests/functional-testing/fns_nac_automation-qa_env.json:7]: After Exporting the environment file from postman, extra space added. But yes, shouldnt be any extra space.

File: api-tests/functional-testing/fns_nac_automation-qa_env.json (hunk 2 above)
Comment (mhaque-ventera) [reply api-tests/functional-testing/fns_nac_automation-qa_env.json:43]: Subscription Key shouldn't be displayed in GitHub.

File: api-tests/functional-testing/fns_nac_automation-postman_collection.json [hunk 7]
```diff
@@ -79,58 +82,277 @@
...
+							"    pm.expect(case_id.length).lessThan(21);\r",
+							"console.log(\"Case ID Length: \"+ JSON.stringify(case_id.length));\r",
+							"    }else if (matchLength == 0) {\r",
+							"        console.log(\"No match found. Case_id not returned\")\r",
+							"    }\r",
+							"})\r",
+							"\r",
+							"pm.test(\"case ID Special Charecter validation\", function(){\r",
+							"     if(matchLength >0){\r",
+							"    const case_id = responseMatches.matches[0].case_id;\r",
```
Comment (dborgen2) [inline api-tests/functional-testing/fns_nac_automation-postman_collection.json:130]: I think this test assumes case_id is not null or empty. We should probably use the regex `/^[0-9a-zA-Z\-_]*$` instead, but I would be curious to see if this test actually passes if case_id is null or empty.

File: api-tests/functional-testing/fns_nac_automation-postman_collection.json [hunk 8]
```diff
@@ -79,58 +82,277 @@
...
+							"    if(matchLength >0){\r",
+							"    const case_id = responseMatches.matches[0].case_id;\r",
+							"    pm.expect(case_id.length).lessThan(21);\r",
+							"console.log(\"Case ID Length: \"+ JSON.stringify(case_id.length));\r",
+							"    }else if (matchLength == 0) {\r",
+							"        console.log(\"No match found. Case_id not returned\")\r",
+							"    }\r",
+							"})\r",
+							"\r",
+							"pm.test(\"case ID Special Charecter validation\", function(){\r",
```
Comment (dborgen2) [inline api-tests/functional-testing/fns_nac_automation-postman_collection.json:128]: Nitpick (non-blocking): Misspelling on "Character".

File: api-tests/functional-testing/fns_nac_automation-postman_collection.json [hunk 9]
```diff
@@ -79,58 +82,277 @@
...
-							"\r",
-							"    }else if(matchLength == 0){\r",
-							"        console.log(\"No match found. participant_id not returned\")\r",
+							"console.log(\"Participant ID Length: \"+ JSON.stringify(participant_id.length));\r",
+							"    }else if (matchLength == 0) {\r",
+							"        console.log(\"No match found. Participant_id not returned\")\r",
 							"    }\r",
-							"\r",
 							"})\r",
 							"pm.test(\"Participant ID Special Charecter validation\", function(){\r",
```
Comment (dborgen2) [inline api-tests/functional-testing/fns_nac_automation-postman_collection.json:146]: Nitpick (Non-Blocking): Misspelling on "Character".

File: api-tests/functional-testing/fns_nac_automation-postman_collection.json [hunk 10]
```diff
@@ -79,58 +82,277 @@
...
+							"     if(matchLength >0){\r",
 							"    const participant_id = responseMatches.matches[0].participant_id;\r",
 							"    pm.expect(/^[0-9a-zA-Z\\-_]+$/.test(participant_id)).to.be.true;\r",
-							"    console.log(\"Participant Response\"+ JSON.stringify(participant_id.length));\r",
+							"console.log(\"Participant ID: \"+ JSON.stringify(participant_id));\r",
+							"\r",
 							"    }\r",
-							
 							"})\r",
+							""
```
Comment (dborgen2) [inline api-tests/functional-testing/fns_nac_automation-postman_collection.json:154]: Question: Should we add any checks for the following:
vulnerable_individual is either null, true, or false.
participant_closing_date is either null, or a string in the format yyyy-mm-dd
recent_benefit_issuance_dates is either an empty list, or if it contains values that each value has a start and end date, each with the format yyyy-mm-dd.
match_url is of the format: https://*/match/{matchId}. The Match URL base path could be set in the environment file perhaps.

File: api-tests/functional-testing/fns_nac_automation-postman_collection.json [hunk 11]
```diff
@@ -79,58 +82,277 @@
...
 							"   }\r",
-							"\r",
+							"    }\r",
 							"})\r",
 							"\r",
 							"pm.test(\"Response time is less than 3000ms\", function () {\r",
 							"    pm.expect(pm.response.responseTime).to.be.below(3000);\r",
 							"});\r",
 							"\r",
+							"const attributesList = [\"match_id\", \"state\", \"case_id\", \"participant_id\", \"participant_closing_date\"];\r",
```
Comment (dborgen2) [inline api-tests/functional-testing/fns_nac_automation-postman_collection.json:107]: I mentioned this to you offline as well, but just to record it on here:
Only match_id, state, participant_id, and match_url are required.
case_id, participant_closing_date, recent_benefit_issuance_dates, and vulnerable_individual are not required. For recent_benefits_issuance_dates, if we are not provided a value it may just return an empty array, in which case for testing purposes it can be treated as mandatory.

Comment (dborgen2) [reply api-tests/functional-testing/fns_nac_automation-postman_collection.json:107]: To that end, we probably want 2 tests here: One for testing that required attributes have a value, and one for testing that all attributes at least come back (even if their value is null)

File: api-tests/functional-testing/fns_nac_automation-qa_env.json (hunk 4 above)
Comment (mhaque-ventera) [reply api-tests/functional-testing/fns_nac_automation-qa_env.json:99]: Added

File: api-tests/functional-testing/fns_nac_automation-postman_collection.json (hunk 6 above)
Comment (mhaque-ventera) [reply api-tests/functional-testing/fns_nac_automation-postman_collection.json:165]: Removed unnecessary comments

File: api-tests/functional-testing/fns_nac_automation-postman_collection.json (hunk 7 above)
Comment (mhaque-ventera) [reply api-tests/functional-testing/fns_nac_automation-postman_collection.json:130]: Updated the code

File: api-tests/functional-testing/fns_nac_automation-postman_collection.json (hunk 10 above)
Comment (mhaque-ventera) [reply api-tests/functional-testing/fns_nac_automation-postman_collection.json:154]: Added validation for the above conditions

File: api-tests/functional-testing/globals.postman_globals.json [hunk 12]
```diff
@@ -36,4 +36,4 @@
 	"_postman_variable_scope": "globals",
 	"_postman_exported_at": "2022-11-11T19:20:03.870Z",
 	"_postman_exported_using": "Postman/10.0.42"
-}
\ No newline at end of file
+}
```
Comment (bgfoshay) [inline api-tests/functional-testing/globals.postman_globals.json:39]: Nitpick: Did you intend to remove the EOL?

File: api-tests/functional-testing/generateRandomDataFile.js [hunk 13]
```diff
@@ -7,4 +8,6 @@ const parsed = parse(file, {
   skip_empty_lines: true
 })
 const chosenRecord = parsed[Math.floor(Math.random() * parsed.length)]
-fs.writeFileSync(outputFilename, JSON.stringify([chosenRecord]))
\ No newline at end of file
+const rowLength = parsed.length;
+chosenRecord["Total_Row_Count"]=rowLength;
+fs.writeFileSync(outputFilename, JSON.stringify([chosenRecord]))
```
Comment (bgfoshay) [inline api-tests/functional-testing/generateRandomDataFile.js:13]: Nitpick: Did you intend to remove the EOL?

File: api-tests/functional-testing/fns_nac_automation-qa_env.json [hunk 14]
```diff
@@ -87,19 +81,19 @@
...
 		}
 	],
 	"_postman_variable_scope": "environment",
-	"_postman_exported_at": "2022-11-11T19:19:48.754Z",
-	"_postman_exported_using": "Postman/10.0.42"
-}
\ No newline at end of file
+	"_postman_exported_at": "2023-01-31T14:47:32.973Z",
+	"_postman_exported_using": "Postman/10.8.7"
+}
```
Comment (bgfoshay) [inline api-tests/functional-testing/fns_nac_automation-qa_env.json:99]: Nitpick: Did you intend to remove the EOL?

File: api-tests/functional-testing/fns_nac_automation-postman_collection.json [hunk 15]
```diff
@@ -279,25 +294,12 @@
 			"response": []
 		},
 		{
-			"name": "Bulk Upload APIM MT-TEST",
+			"name": "Bulk Upload APIM EA-TEST",
```
[the remark this answers is not in the document]
Comment (mhaque-ventera) [reply api-tests/functional-testing/fns_nac_automation-postman_collection.json:297]: I am running these tests on Test Environment only. But yes, if other members run these tests in the Test env there will be conflict.

[the remark this answers is not in the document]
Comment (mhaque-ventera) [reply api-tests/functional-testing/fns_nac_automation-postman_collection.json:297]: And Ryan suggested me to use EA, EB, EC for Test Env.

[the remark this answers is not in the document]
Comment (bgfoshay) [reply api-tests/functional-testing/fns_nac_automation-postman_collection.json:297]: In that case I think maybe we should parameterize the states. EA, EB, EC might be fine for Test Env, but ideally we should be able to run this suite of tests against any env. I'll follow up with you offline

File: api-tests/functional-testing/fns_nac_automation-postman_collection.json [hunk 16]
```diff
@@ -714,12 +501,44 @@
...
+							"pm.test(\"Response time is less than 3000ms\", function () {\r",
+							"    pm.expect(pm.response.responseTime).to.be.below(3000);\r",
+							"});\r",
+							"\r",
+							"\r",
+							"pm.test(\"Verify Mandatory attributes Has value\", () => {\r",
+							"    for (const match of required_attributesListIfMatch) {\r",
+							"        if (matchLength > 0) {\r",
+							"            for (let i = 0; i < matchLength; i++) {\r",
+							"                pm.expect(responseMatches.matches[i].match).not.to.be.eq(null);\r",
```
Comment (dborgen2) [inline api-tests/functional-testing/fns_nac_automation-postman_collection.json:524]: If the property has an empty string value I think it will pass this test. I think it should be checking for not null or empty.

File: api-tests/functional-testing/fns_nac_automation-postman_collection.json [hunk 17]
```diff
@@ -749,113 +568,214 @@
...
-							"        for (const matches of matchereResponse) {\r",
-							"            count[matches] = count[matches] ? count[matches] + 1 : 1;\r",
-							"        }\r",
-							"        for (var k = 0; k < matchereResponse.length; k++) {\r",
-							"            pm.expect(count[matchereResponse[1]]).to.be(1);\r",
-							"        }\r",
+							"    matchereResponse = responseMatches.matches.map((match) => {\r",
+							"        return match.match_id;\r",
+							"    })\r",
+							"    if (matchereResponse.length > 0) {\r",
```
Comment (dborgen2) [inline api-tests/functional-testing/fns_nac_automation-postman_collection.json:577]: At this point in the test we have only uploaded EA, correct? Assuming unique LDS hashes (because they were randomly/timed generated), we should not have any matches because we are finding matches as EA.

I think the only tests we need here is the "Status code validation" test, the "Response time is less than 3000ms" test, and a new test that makes sure that matchLength == 0. Is this correct?

File: api-tests/functional-testing/fns_nac_automation-postman_collection.json [hunk 18]
```diff
@@ -143,10 +143,10 @@
 							"        console.log(\"No match found. Participant_id not returned\")\r",
 							"    }\r",
 							"})\r",
-							"pm.test(\"Participant ID Special Charecter validation\", function(){\r",
+							"pm.test(\"Participant ID Special character validation\", function(){\r",
 							"     if(matchLength >0){\r",
 							"    const participant_id = responseMatches.matches[0].participant_id;\r",
-							"    pm.expect(/^[0-9a-zA-Z\\-_]+$/.test(participant_id)).to.be.true;\r",
+							"    pm.expect(/^[0-9a-zA-Z\\-_]*$/.test(participant_id)).to.be.true;\r",
```
Comment (dborgen2) [inline api-tests/functional-testing/fns_nac_automation-postman_collection.json:149]: At this point in the test we haven't uploaded anything, correct? Assuming unique LDS hashes (because they were randomly/timed generated), we should not have any matches.

I think the only tests we need here is the "Status code validation" test, the "Response time is less than 3000ms" test, and a new test that makes sure that matchLength == 0. Is this correct?

File: api-tests/functional-testing/fns_nac_automation-postman_collection.json [hunk 19]
```diff
@@ -724,127 +906,868 @@
...
+							"            }\r",
+							"        }\r",
+							"\r",
+							"    }\r",
+							"})\r",
+							"\r",
+							"pm.test(\"Finding Match betwen state \", function () {\r",
+							"    for (let i = 0; i < matchLength; i++) {\r",
+							"        const stateT = responseMatches.matches[i].state;\r",
+							"        if (initiating_state === \"EC\") {\r",
```
Comment (dborgen2) [inline api-tests/functional-testing/fns_nac_automation-postman_collection.json:1093]: General comment for Find Matches APIM EB-TEST:
We should only have matches for EA, and our intiating state is EB. I think our tests here should make sure that we are only getting back matches for EA, and no matches for EB or EC or MT

File: api-tests/functional-testing/fns_nac_automation-postman_collection.json [hunk 20]
```diff
@@ -959,180 +1909,378 @@
...
```
Comment (dborgen2) [inline api-tests/functional-testing/fns_nac_automation-postman_collection.json:2025]: General comment for Find Matches APIM EC-TEST:
We should only have matches for EA and EB, and our initiating state should always be EC. I think our tests here should make sure that we are only getting back matches for EA and EB, and no matches for EC or MT

File: api-tests/functional-testing/fns_nac_automation-postman_collection.json [hunk 21]
```diff
@@ -724,127 +906,868 @@
...
+							"console.log(\"upload_id: \"+upload_id);"
+						],
+						"type": "text/javascript"
+					}
+				},
+				{
+					"listen": "test",
+					"script": {
+						"exec": [
+							"pm.test(\"upload_identifier that is too long is rejected\", function () {\r",
```
Comment (dborgen2) [inline api-tests/functional-testing/fns_nac_automation-postman_collection.json:1722]: Question: Where is this upload_id coming from? Is it the uploadId that we got during the last bulk upload? If so, how would it ever have a length > 1000 or have invalid characters?

If that is the case, you may need to hardcode the uploadId when calling this test and break the test into two tests

File: api-tests/functional-testing/fns_nac_automation-postman_collection.json [hunk 22]
```diff
@@ -959,180 +1909,378 @@
...
+						],
+						"type": "text/javascript"
+					}
+				},
+				{
+					"listen": "test",
+					"script": {
+						"exec": [
+							"\r",
+							"pm.test(\"Status code validation for Calling the API without an identifier\", function () {\r",
```
Comment (dborgen2) [inline api-tests/functional-testing/fns_nac_automation-postman_collection.json:1930]: Minor nitpick: I think the name of this function is incorrect. The identifier is 00000000000000000000, so it has an identifier but the identifier just doesn't exist.

File: api-tests/functional-testing/fns_nac_automation-postman_collection.json [hunk 23]
```diff
@@ -1237,60 +2382,84 @@
...
-							"    const participant_id = responseMatches.matches[0].participant_id;\r",
-							"    pm.expect(participant_id.length).lessThan(21);\r",
-							"    console.log(\"Participant Response\"+ JSON.stringify(participant_id.length));\r",
+							"const attributesList = [\"match_id\", \"state\", \"case_id\", \"participant_id\", \"participant_closing_date\",\"recent_benefit_issuance_dates\",\"vulnerable_individual\",\"match_url\"];\r",
+							"const mandatoryAttributesList = [\"match_id\", \"state\", \"participant_id\",\"match_url\"];\r",
+							"pm.test(\"Validate Presence of attributes In response \", ()=>{\r",
 							"\r",
-							"    }else if(matchLength == 0){\r",
-							"        console.log(\"No match found. participant_id not returned\")\r",
+							"    if(matchLength >0){\r",
```
Comment (dborgen2) [inline api-tests/functional-testing/fns_nac_automation-postman_collection.json:2411]: Minor nitpick: Just a general comment, but the indentation of these files seem off both here and when I import them to Postman. Is this something that gets messed up when Postman exports, or can it be fixed?

File: api-tests/functional-testing/fns_nac_automation-postman_collection.json [hunk 24]
```diff
@@ -1364,21 +2532,21 @@
 							"    console.log(\"Current length of Results: \"+matchLength);\r",
 							"    pm.expect(matchLength).to.greaterThan(0);\r",
```
Comment (dborgen2) [inline api-tests/functional-testing/fns_nac_automation-postman_collection.json:2533]: Question: If possible, it would be nice if we could test the exact number of matches expected instead of just greater than 0. Is this possible?

File: api-tests/functional-testing/fns_nac_automation-postman_collection.json [hunk 25]
```diff
@@ -197,7 +215,6 @@
 					"listen": "test",
 					"script": {
 						"exec": [
-							
 							"const initiating_state = pm.request.headers.get(\"X-Initiating-State\");\r",
 							"\r",
 							"pm.test(\"Status code validation\", function () {\r",
```
Comment (dborgen2) [inline api-tests/functional-testing/fns_nac_automation-postman_collection.json:220]: Question: I see we have a test here for missing participant ID, but what about if we are missing LDS hash or Search Reason? Or if Lds Hash, Case Id, Participant Id, or Search Reason are in the wrong format? Maybe those are not necessary in this first round of testing, but thought I should check.

Comment (bgfoshay) [reply api-tests/functional-testing/fns_nac_automation-postman_collection.json:220]: It's a good question, but Masud has been incrementally going through the UAT tests defined in a Word doc by PMs. The tests you suggested should be covered as well. He and I spoke and he's going to create a separate ticket to cover any of these tests that aren't already listed in the Word doc.

File: api-tests/functional-testing/fns_nac_automation-postman_collection.json (hunk 15 above)
[the remark this answers is not in the document]
Comment (bgfoshay) [reply api-tests/functional-testing/fns_nac_automation-postman_collection.json:297]: After discussion on this, Masud noted that he has variables for the states but the names are hard-coded which could be confusing e.g. stateEA variable could be set to "MT". Creating a separate ticket to clean this up.

File: api-tests/functional-testing/fns_nac_automation-postman_collection.json (hunk 23 above)
Comment (bgfoshay) [reply api-tests/functional-testing/fns_nac_automation-postman_collection.json:2411]: I was wondering the same thing. It'd be nice if we could figure out a way to pretty-print the contents of the file on export so that the indentation and formatting stay consistent. That'll make readability better and review easier. I'll add a separate ticket to look into this.

File: api-tests/functional-testing/fns_nac_automation-postman_collection.json (hunk 16 above)
Comment (mhaque-ventera) [reply api-tests/functional-testing/fns_nac_automation-postman_collection.json:524]: pm.expect(responseMatches.matches[i].match).not.to.be.eq(null); -> this specific line will make sure the value is not null.

**Review by ccaprio97 — APPROVED** [review]
These changes look good to merge. Suggested enhancements and additional tests identified during the review will be addressed in subsequent JIRA ticket.
